packages/shared/src/guards.ts

```diff
@@ -421,34 +421,37 @@ export function isComplexValue(
   return type.class === DTypeClass.Complex;
 }
 
-function assertPrimitiveValue<D extends Dataset>(
-  dataset: D,
+function assertPrimitiveValue<T extends DType>(
+  type: T,
```

Comment:
Unrelated refactoring here: I'm just opening the door to asserting compound types more strictly in the future.